### PR TITLE
remove "doc" make target and rename "user_doc"

### LINKED ISSUE #2874: remove "doc" make target and rename "user_doc"
The other day I wanted to verify that a change to the docs rendered correctly. So I ran `make doc` and was surprised to find the resulting web page looked nothing like that on fishshell.com. Turns out the correct command is `make user_doc`. The "doc" target is meant to build the documentation for the source code. Since it was recently decided that we would no longer require Doxygen markup of the source and have an open cleanup issue to remove it we should remove the `make doc` command. While we're at it we might as well rename the "user_doc" target to "doc".

### PR DESCRIPTION
**Pull request by krader1961**
Fixes #2874

### CODE REVIEW

**Comment by zanchey** [comment]
The other file that needs fixing is `build_tools/make_tarball.sh`, which is used for building integration builds and releases.

File: Makefile.in [hunk 1]
```diff
@@ -266,29 +266,23 @@ prof: all
...
-	(cat Doxyfile.user; echo INPUT_FILTER=./lexicon_filter; \
-	echo PROJECT_NUMBER=$(FISH_BUILD_VERSION) | $(SED) "s/-.*//") | doxygen - && touch user_doc; \
-	cd user_doc/html && rm -f bc_s.png bdwn.png closed.png ftv2*.png nav*.png open.png sync_*.png tab*.* doxygen.* dynsections.js jquery.js pages.html
-
-#
-# Source code documentation. Also includes user documentation.
+# Depend on the sources (*.hdr.in) and manually make the intermediate *.hdr
+# and doc.h files if needed. The sed command deletes everything including and
+# after the first -, for simpler version numbers Cleans up the user_doc/html
+# directory once Doxygen is done.
```
Comment (zanchey) [inline Makefile.in]: Did this get accidentally reflowed?

**Comment by krader1961** [comment]
> The other file that needs fixing is build_tools/make_tarball.sh,

Fixed. Also found a reference in *doc_src/FORMATTING.md*.

File: Makefile.in (hunk 1 above)
Comment (krader1961) [inline Makefile.in]: The reflow was intentional. This wasn't really a bulleted list. And where a bulleted list is intended it should be explicit. Not to mention the weird lack of periods on two of the three sentences.

**Comment by krader1961** [comment]
Fixed via merge of commit 35cee1e39c0ca6bd35f2bb0c18ad6ffca5cc7108

**Comment by floam** [comment]
Why did we rename the target name but not the produced directory? Shouldn't they line up if they can?

We don't have developer docs anymore, seems best to output to `./doc`.

**Pull request closed without merging**